manifests/extension.pp

```diff
@@ -42,6 +49,8 @@
   $provider          = undef,
   $source            = undef,
   $pecl_source       = undef,
+  $so_name           = $name,
+  $php_api_version   = undef,
```

Comment:
We should probably move the `$php_api_version` parameter into the main class, because that setting would apply to all php extensions as we don't support multiple versions of PHP on a system. In case the api version needs to be overridden, every extension hash in hiera would need to specify `php_api_version` which is a bit redundant.

So my suggestion would be to move `$php_api_version` to the `php` class as `$api_version` and use `${::php::api_version}` where `$module_path` is built.